main/integration-tests/__test__/api-tests/workspace-service-catalogs/create-workspace-service-catalog.test.js

```diff
@@ -166,4 +179,90 @@ describe('Create workspace-service-catalog scenarios', () => {
       });
     });
   });
+  describe('Creating a workspace service catalog environment with studies', () => {
+    it('for EC2Linux', async () => {
+      // const researcher1Session = await setup.createResearcherSession();
+      // Things to do
+      // 1. Create user
+      const admin1Session = await setup.createAdminSession();
+      const username = setup.gen.username();
+      const defaultUser = admin1Session.resources.users.defaults({ username });
+      await expect(admin1Session.resources.users.create(defaultUser)).resolves.toMatchObject({
+        username,
+      });
+      // 2. Assign Project ID to user (assigned by default)
+
+      // 3. Create My Study, Org Study, Data Source study
+      const studyIds = [];
+      let studyId = setup.gen.string({ prefix: `create-study-ray-my-study` });
+      await expect(
+        admin1Session.resources.studies.create({ id: studyId, name: studyId, category: 'My Studies' }),
+      ).resolves.toEqual(
+        expect.objectContaining({
+          id: studyId,
+        }),
+      );
+      studyIds.push(studyId);
+
+      studyId = setup.gen.string({ prefix: `create-study-ray-org-study` });
+      await expect(
+        admin1Session.resources.studies.create({ id: studyId, name: studyId, category: 'Organization' }),
+      ).resolves.toEqual(
+        expect.objectContaining({
+          id: studyId,
+        }),
+      );
+      studyIds.push(studyId);
+
+      // 4. Create workspace with the above studies
+      const workspaceName = setup.gen.string({ prefix: 'workspace-sc-test' });
+
+      const env = await admin1Session.resources.workspaceServiceCatalogs.create({
+        name: workspaceName,
+        envTypeId: setup.defaults.envTypes.ec2Linux.envTypeId,
+        envTypeConfigId: setup.defaults.envTypes.ec2Linux.envTypeConfigId,
+        studyIds,
+        description: 'assignment',
+        projectId: setup.defaults.project.id,
+        cidr: '54.240.196.186/32',
+      });
+      expect(env).toEqual(
+        expect.objectContaining({
+          name: workspaceName,
+          envTypeId: setup.defaults.envTypes.ec2Linux.envTypeId,
+          envTypeConfigId: setup.defaults.envTypes.ec2Linux.envTypeConfigId,
+          studyIds,
+        }),
+      );
+
+      const workflows = await admin1Session.resources.workflows
+        .versions('wf-provision-environment-sc')
+        .version(1)
+        .instances()
+        .get();
+      let result;
+      for (const w of workflows) {
+        if (
+          w.input.envId === env.id &&
+          w.input.envTypeId === env.envTypeId &&
+          w.input.envTypeConfigId === env.envTypeConfigId
+        ) {
+          result = w;
+        }
+      }
+      while (result.wfStatus !== 'done' && result.wfStatus !== 'error') {
```

Comment:
Check out other [tests with workflows](https://github.com/awslabs/service-workbench-on-aws/tree/mainline/main/integration-tests/__test__/api-tests/workflows), specifically `get-workflow-instances.test.js`. Most of these kinds of things we already have helper methods for, in this case I'm pretty sure `.triggerAndWait()` does the same thing as this while loop but slightly cleaner.

Reply:
`triggerAndWait()` can't be used for my test, because the workflow is triggered when the workspace is provisioned. I looked into implementing the waiting aspect for that function, but I still obtain the no-await-in-loop error. Do you happen to know why `triggerAndWait()` doesn't run into this issue itself?

Reply:
I'm not super familiar with the method, but if I had to guess it's probably because `triggerAndWait()` doesn't pause execution, it just returns a `Promise()` for the value that's unresolved until the function completes without requiring the loop to wait.

Reply:
> `triggerAndWait()` can't be used for my test, because the workflow is triggered when the workspace is provisioned. I looked into implementing the waiting aspect for that function, but I still obtain the no-await-in-loop error. Do you happen to know why `triggerAndWait()` doesn't run into this issue itself?

Are you getting linting warnings?  if so, you can suppress them, we do where ever it is appropriate.

Reply:
Yes, I started suppressing them. A new issue is that the code will seem to run multiple times if I use lodash or a while-loop based polling technique. For example, one test run should provision one workspace and complete one workflow accordingly, but it will provision several workspaces and workflows until it halts. I believe this issue is from the async/await being in a loop. Is there a better technique for polling the workflow's status?

Reply:
Some javascript constructs support the usage of async (depending on node.js version)  for example 'for  of'  and 'while' support async/await.  lodash _.forEach() does not support it and array.forEach() does not support async.  It is possible that you might have used a combination that does not support async/await.

Reply:
It turns out that the multiple workspaces bug was the Jest retry logic, it doesn't print out errors until the last retry fails. I also changed the retry count to 1 because workspace provisioning has a large time overhead.